crates/extensions/c8y_auth_proxy/src/server.rs

```diff
@@ -0,0 +1,307 @@
+use crate::tokens::*;
+use axum::body::Body;
+use axum::body::Full;
+use axum::body::StreamBody;
+use axum::extract::FromRef;
+use axum::extract::State;
+use axum::http::uri::PathAndQuery;
+use axum::http::HeaderValue;
+use axum::response::Response;
+use axum::routing::get;
+use axum::routing::IntoMakeService;
+use axum::Router;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use hyper::server::conn::AddrIncoming;
+use hyper::HeaderMap;
+use hyper::Uri;
+use miette::Context;
+use miette::IntoDiagnostic;
+use reqwest::Method;
+use reqwest::StatusCode;
+use std::fmt;
+use std::net::IpAddr;
+use std::sync::Arc;
+use tracing::info;
+
+type AxumServer = hyper::Server<AddrIncoming, IntoMakeService<Router>>;
+
+pub struct Server {
+    fut: BoxFuture<'static, hyper::Result<()>>,
+}
+
+impl Server {
+    pub(crate) fn try_init(state: AppState, address: IpAddr, port: u16) -> miette::Result<Self> {
+        Ok(Server {
+            fut: try_run_server(address, port, state)?.boxed(),
+        })
+    }
+
+    pub fn wait(self) -> BoxFuture<'static, hyper::Result<()>> {
+        self.fut
+    }
+}
+
+fn try_run_server(address: IpAddr, port: u16, state: AppState) -> miette::Result<AxumServer> {
+    info!("Launching on port {port}");
+    let handle = get(respond_to)
+        .post(respond_to)
+        .put(respond_to)
+        .patch(respond_to)
+        .delete(respond_to)
+        .options(respond_to);
+    let app = Router::new()
+        .route("/", handle.clone())
+        .route("/*path", handle)
+        .with_state(state);
```

Comment:
So the current proposal is to proxy `https://didier.latest.stage.c8y.io` directly on `http://local-thin-edge-device/`.

I would consider to add some `c8y` prefix. The idea being to use the same HTTP server to serve local file-transfer requests.

@reubenmiller @albinsuresh What do you think here.

Reply:
Yes having a `c8y` route makes sense, and it would be consistent to the way the mapper specific topics work, and allow for extending it to other cloud platforms in the future (if similar API exists).

Reply:
Agree that it's good to have that prefix to support different services in future as well.

But, question on the prospect of using the same server for file-transfer service as well: The file-transfer service is a generic cloud-agnostic service hosted rightly by tedge-agent, and **not** the mapper, right? And this service and any future cloud proxy services belongs to their own respective mappers. So, how can we combine both?